Automattic/jetpack
# [Boost] Connect image guide ui to back-end

Connect the Image Guide UI to the back-end. Requires D106989-code on the dotcom side.

This is gated behind a feature-flag. This PR is also getting a bit long in the tooth and really needs to land soon. So, I recommend that we work through critical bugs in this PR through review - but non-critical bugs should get turned into a ticket on our current focus board to get fixed after this PR is merged.

## Proposed changes:
* Hook up boost image guide ui to back-end
* Clean out some of the status/ignore stuff we are excluding from the first release
* Add translation markers on user-facing strings. 
* Add options to start image guide generation.
* Make the image guide progress display work.

### Other information:

## Jetpack product discussion
n/a


## Does this pull request change what data or activity we track or use?
no

## Testing instructions:
- Apply D106989-code to your dotcom sandbox.
- Setup `/etc/hosts` to point `public-api` at your sandbox.
- Run Boost Cloud locally, to ensure it directs shield requests to your sandbox by way of your hosts file.
- Ensure your test site has Boost Premium to enable the premium guide
- Go to your boost dashboard, and click to generate an Image Size Analysis report.
- Ensure the UI shows the report in progress, then that you can view the report.
